### PR TITLE
WIP | Full KYB process workflow definition

### PR DESCRIPTION
**Pull request by tomer-shvadron**
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+          CONTINUE: [{ target: 'run_vendor_data_collection' }],
+          FAILED: [{ target: 'auto_reject' }],
+        },
+      },
+      run_hk_kyb: {
+        tags: [StateTag.PENDING_PROCESS],
+        on: {
+          // CONTINUE: [{ target: 'run_aml' }],
+          CONTINUE: [{ target: 'run_vendor_data_collection' }],
+          FAILED: [{ target: 'auto_reject' }],
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts]: It shouldn't go to auto_reject, it should move to "failure" state from there we should implement recovery actions

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    predictableActionArguments: true,
+    initial: 'data_collection',
+    context: {
+      documents: [],
+    },
+    states: {
+      data_collection: {
+        tags: [StateTag.COLLECTION_FLOW],
+        on: {
+          start: [
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts]: "COLLECTION_FLOW_FINISHED"

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+          FAILED: [{ target: 'auto_reject' }],
+        },
+      },
+      run_vendor_data_collection: {
+        tags: [StateTag.PENDING_PROCESS],
+        on: {
+          CONTINUE: [{ target: 'manual_review' }],
+          FAILED: [{ target: 'auto_reject' }],
+        },
+      },
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts:71]: Missing KYC for collected UBO's

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+      // },
+      manual_review: {
+        tags: [StateTag.MANUAL_REVIEW],
+        on: {
+          approve: 'approved',
+          reject: 'rejected',
+          revision: 'revision',
+        },
+      },
+      revision: {
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts:95]: We can leave revision, no need to pending resubmisstion

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+        tags: [StateTag.REVISION],
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts]: add "COLLECTION FLOW" tag

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+        always: [
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts]: Should be explict event name

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+          {
+            target: 'pending_resubmission',
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts]: -> manual_review

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+      auto_reject: {
+        tags: [StateTag.REJECTED],
+        type: 'final' as const,
+      },
+    },
+  },
+  extensions: {
+    apiPlugins: [
+      {
+        name: 'asia_verify_cn_kyb',
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts:122]: Talk Omri about how its saved in the context, lets refactor it a bit if possible

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+        pluginKind: 'kyb',
+        url: `${env.UNIFIED_API_URL}/companies-v2`,
+        method: 'GET',
+        stateNames: ['run_cn_kyb'],
+        successAction: 'CN_KYB_DONE',
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts:127]: Fix actions names

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+              transformer: 'jmespath',
+              mapping: '@', // jmespath
+            },
+          ],
+        },
+      },
+      {
+        // @TODO: handle callback
+        name: 'asia_verify_hk_kyb',
+        pluginKind: 'webhook',
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts]: Its not webhook, have a look a kyc session plugin

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+        method: 'GET',
+        stateNames: ['run_hk_kyb'],
+        headers: { Authorization: 'Bearer {secret.UNIFIED_API_TOKEN}' },
+        request: {
+          transform: [
+            {
+              // @TODO: Work out "entity.data.registrationNumber" with Daniel
+              transformer: 'jmespath',
+              mapping: `{
+                companyNumber: entity.data.registrationNumber,
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts:166]: add callback URL

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+                countryOfIncorporation: 'hk',
+                vendor: 'asia-verify'
+              }`, // jmespath
+            },
+          ],
+        },
+        response: {
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts:197]: Currently, we are not using the response (its the persistence is done from /hooks endpoint)

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 13]
```diff
@@ -0,0 +1,310 @@
...
+            {
+              transformer: 'jmespath',
+              mapping: '@', // jmespath
+            },
+          ],
+        },
+      },
+      {
+        name: 'asia_verify_aml',
+        pluginKind: 'aml',
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts]: move to pluginkind API

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 14]
```diff
@@ -0,0 +1,310 @@
...
+        url: `${env.UNIFIED_API_URL}/companies`,
```
Comment (Blokh) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts]: {secret.UNIFIED_API_URL}/companies/{entity.data.additionalInfo.country}/{entity.data.companyName}/aml

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 15]
```diff
@@ -0,0 +1,310 @@
...
+        name: 'resubmission_email',
+        pluginKind: 'email',
+        url: `{secret.EMAIL_API_URL}`,
+        method: 'POST',
+        stateNames: ['pending_resubmission'],
+        headers: {
+          Authorization: 'Bearer {secret.EMAIL_API_TOKEN}',
+          'Content-Type': 'application/json',
+        },
+        request: {
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts:248]: ITs okay for Thursday, but add a task for next sprint and make sure its pulled into the sprint

File: services/workflows-service/scripts/workflows/full_kyb_process.ts [hunk 16]
```diff
@@ -0,0 +1,310 @@
...
+          transform: [
+            {
+              transformer: 'jmespath',
+              mapping: `{
+                kybCompanyName: entity.data.companyName,
+                customerCompanyName: entity.data.additionalInfo.ubos[0].entity.data.additionalInfo.customerCompany,
+                firstName: entity.data.additionalInfo.mainRepresentative.firstName,
+                resubmissionLink: join('',['https://',entity.data.additionalInfo.ubos[0].entity.data.additionalInfo.normalizedCustomerCompany,'.demo.ballerine.app','/workflowRuntimeId=',workflowRuntimeId,'?resubmitEvent=RESUBMITTED']),
```
Comment (alonp99) [inline services/workflows-service/scripts/workflows/full_kyb_process.ts:256]: Lior: Ill send you env var for the subdomain

**Pull request closed without merging**